### ISSUE TITLE
Staging Review - Accessibility Feedback - Design System Team - File Input web component

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#46990]
## VFS acceptance criteria
- [ ] Let Platform know when the **Must** feedback has been incorporated 
- [ ] Leave any comments for feedback you decide _not_ to take
- [ ] VFS team closes the ticket

## Feedback
- Practice areas will document their feedback following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback).

**Must:**
I'm getting some inconsistent behavior on NVDA and JAWS. Example:

```
<VaFileInput
  accept="image/*"
  button-text="Upload file"
  error="We couldn't upload your file"
  label="Upload additional evidence"
  name="my-file-input"
  onVaChange={function noRefCheck(){}}
  required
/>
```

... which renders as:

```
<va-button id="fileInputButton" aria-label="Upload additional evidence" aria-describedby="error-message" secondary="" class="hydrated"></va-button>
```

... and in the shadow DOM is:

```
<button type="button" part="button">Upload file</button>
```

When I tab to the upload button with JAWS running, the button text, aria-label, and aria-describedby error message are all announced. When I tab to it with NVDA running, just the button text "Upload file" is announced --- no label, no error message.

My best guess is it's because the ARIA attributes are all on the web component but not on the `<button>` in the shadow DOM. JAWS is smart enough to figure out what's happening with the web component, but NVDA maybe isn't.

**Should:**

**Consider:**

## Platform directions
- Remove the governance-team label
- Update "Issue Title"
- Link to collab cycle Request epic
- Add your feedback
- Add assignees based on collab cycle touchpoint
  - **Design Intent**: VFS designer, VFS PM (optional), yourself (optional)
  - **Midpoint Review**: VFS PM, yourself (optional)
  - **Staging Review QA Only**: VFS PM, yourself (optional)

**Comment by jamigibbs** [comment]
Thanks for your feedback on this @briandeconinck! 

I'm trying to reproduce the issue and I'm wondering if part of the problem is that the Storybook example for error isn't "dynamic". What I mean by that is that the error message isn't triggered by anything. I did a quick test in Storybook to always trigger the error when a file is uploaded and this is my result (you can skip to 20 seconds in the video):

https://user-images.githubusercontent.com/872479/190158083-ca04fc20-4ef9-4eff-b4e9-b7384996ed1e.mov

It is reading out the label "Upload additional evidence", identifying the button to that label, and after the file is uploaded it reads the error message as "Error: We couldn't upload your file".

I'm testing this in Browserstack with their beta integration of NVDA so it's possible that we have differences there.

**Comment by briandeconinck** [comment]
@jamigibbs Sorry for the slow reply, and thank you for looking at this! From the video I agree, it seems to be announcing as intended. I'm a *little* concerned that there's still some weirdness happening that's not being exposed through the Browserstack + NVDA integration... but since JAWS announced fine when I tested, I'm fine with this. As the component gets used I'll keep an eye out during Collab Cycle reviews to verify.

**Comment by jamigibbs** [comment]
Thanks @briandeconinck and I appreciate your eyes there. As a sanity check, I'm going to do a little more testing on my end with it on a vets-website page (not just in Storybook). If I'm still not able to reproduce any issues, I'll close this ticket but I appreciate you keeping an eye on it during collab cycle reviews too.
